### PR TITLE
Support peek_next_page and skip_next_page in InMemoryPageReader

### LINKED ISSUE #2406: Support peek_next_page and skip_next_page in InMemoryPageReader
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
when i was implementing bench using `skip_records` got

```
Benchmarking arrow_array_reader/Int32Array/binary packed skip, mandatory, no NULLs: Warming up for 3.0000 sthread 'main' panicked at 'not implemented', /CLionProjects/github/arrow-rs/parquet/src/util/test_common/page_util.rs:169:9

```

which is unimplemented

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?

Closes #2406.
Related #2393 

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by Ted-Jiang** [comment]
@tustvold ptal

File: parquet/src/util/test_common/page_util.rs [hunk 1]
```diff
@@ -166,11 +167,28 @@ impl<P: Iterator<Item = Page> + Send> PageReader for InMemoryPageReader<P> {
     }
 
     fn peek_next_page(&mut self) -> Result<Option<PageMetadata>> {
-        unimplemented!()
+        if let Some(x) = self.page_iter.peek() {
+            match x {
+                Page::DataPage { .. } => {
+                    unreachable!()
```
Comment (tustvold) [inline parquet/src/util/test_common/page_util.rs:173]: Why is this unreachable?

Comment (Ted-Jiang) [reply parquet/src/util/test_common/page_util.rs:173]: I found all test pages read by InMemoryPageReader are using DataPagev2.
and DataPagev1 not have the 'num_rows' info.

Comment (Ted-Jiang) [reply parquet/src/util/test_common/page_util.rs:173]: I my mind , i think most application still use v1 page(which not supported `DeltaBitPack`). Is iox using datapagev2?

Comment (tustvold) [reply parquet/src/util/test_common/page_util.rs:173]: Yeah, most applications have yet to update to parquet v2 despite it being released almost a decade ago now :laughing: 

IOx isn't currently, but may in future. DeltaBitPack is very good for storing sorted timestamps as are common for our workload. Although their are some questionable decisions in the DeltaBitPack encoding scheme that make it more expensive than it really should be... Perhaps that is why most applications haven't bothered to upgrade :sweat_smile: 

That was why I was surprised to see this isn't being exercised :sweat_smile:

Comment (Ted-Jiang) [reply parquet/src/util/test_common/page_util.rs:173]: Cool ! thanks for your info.

Comment (Ted-Jiang) [reply parquet/src/util/test_common/page_util.rs:173]: >I can't remember if column index support requires v2 🤔

supported both v1 and v2

**Review by tustvold — APPROVED** [review]

**Pull request merged**